web/src/overlay_util.ts

```diff
@@ -13,3 +13,32 @@ export function disable_scrolling(): void {
 export function enable_scrolling(): void {
     $("html").css({"overflow-y": "scroll", "--disabled-scrollbar-width": "0px"});
 }
+
+export function trapFocus(overlay: JQuery): void {
+    overlay.on("keydown", (e: JQuery.KeyDownEvent) => {
+        // Collect all elements that can be focused using the Tab key.
+        const tabbableElements = overlay
+            .find("input, select, textarea, a, [href], button, [tabindex]:not([tabindex='-1'])")
+            .filter(":visible")
+            .not(":disabled");
+
+        if (tabbableElements.length === 0) {
+            return;
+        }
+
+        const firstElement = tabbableElements.first();
+        const lastElement = tabbableElements.last();
+
+        // 1. If the focus is on the first element and Shift + Tab is pressed, move the focus to the last element.
+        // 2. If the focus is on the last element and Tab is pressed, move it to the first element.
+        // Thus, trapping the focus inside the overlay.
+        if (e.key === "Tab") {
+            const isShiftTab = e.shiftKey && $(e.target).is(firstElement);
+            const isTab = !e.shiftKey && $(e.target).is(lastElement);
+            if (isShiftTab || isTab) {
+                e.preventDefault();
+                (isShiftTab ? lastElement : firstElement).trigger("focus");
+            }
+        }
+    });
+}
```

Comment:
Did you look at #27135? That seems like a less messy implementation of this concept.

Reply:
Yeah I have seen that. But I thought it would be better if the logic is concentrated in the overlay itself, as the previous implementation might need adding the `<span>` at bottom of every overlay that's present now and maybe if any in future?

Though, I can modify my implementation like #27135, if you think that's a better approach.

Reply:
I think fetching all the elements is slower than adding just two event listeners and destroying them while closing the overlays. I have updated my implementation, please have a look.

Reply:
Adding two spans at the top and bottom, means, I have to press `Tab` key two times extra for the focus to shift from last element to first element. So, that might be something to consider.

Also, currently, after the last element in right panel, focus shifts to left sidebar next, but there is no indicator for that on "Personal" and "Organization" tabs at the top.

Reply:
But, isn't it so confusing for the developer to choose the first or last focus trapper from the already existing elements? Because in many areas, the focus-able elements are being rendered dynamically in the DOM through JS, so wouldn't it be difficult to locate the first, last elements of every template by searching in `web/src`?


Though, I have added indicator for the **Personal** and **Organization** tabs in my latest code.